types/rethinkdb/index.d.ts

```diff
@@ -527,6 +528,21 @@ declare module "rethinkdb" {
         append(prop: string): Expression<Object>;
         contains(prop: string): Expression<boolean>;
 
+        /**
+         * for filter, will be used as Expression<boolean>
+         * for selector, will be used as projector
+         */
+        match(regexp: string): Expression<{
+            start: number,
+            end: number,
+            str: string,
+            groups: Array<{
+                start: number,
+                end: number,
+                str: string
+            }>
+        }> | Expression<boolean>;
```

Comment:
This function also returns `null`, not boolean when nothing is matched.
https://www.rethinkdb.com/api/javascript/#match